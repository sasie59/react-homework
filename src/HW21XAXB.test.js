import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import HW,{checkAb,makeRandom,isErrorValue} from './HW21XAXB';

test('should wrong format', () => {
  const { container, getByTestId } = render(<HW />);
  const input = container.querySelector('input');
  userEvent.type(input, '1111{enter}');
  const errMsg = getByTestId('errMsg');
  expect(errMsg).toBeTruthy();

  userEvent.type(input, 'aqws{enter}');
  expect(errMsg).toBeTruthy();
});

test('should ringt input', () => {
  const { container, getByTestId } = render(<HW />);
  const input = container.querySelector('input');

  userEvent.type(input, '1234{enter}');
  expect(getByTestId('result-0').textContent).toBe('1234: 0A1B');

  userEvent.type(input, '7406{enter}');
  expect(getByTestId('result-0').textContent).toBe('1234: 1A0B');
});

test('should clear arr', () => {
  const { container, getByTestId } = render(<HW />);
  const input = container.querySelector('input');
  userEvent.type(input, '1234{enter}');
  expect(getByTestId('result-0').textContent).toBe('1234: 1A1B');
  userEvent.type(input, '2345{enter}');
  expect(getByTestId('result-1').textContent).toBe('2345: 1A1B');
  userEvent.type(input, '2137{enter}');
  expect(getByTestId('bingo').textContent).toBe('bingo ! game over ， replay?Yes');
  expect(getByTestId('ul').children.length).toBe(1);
});

test('should bingo Game Over & new Game ', () => {
  const { container,getByTestId } = render(<HW />);
  const input = container.querySelector('input');
  userEvent.type(input, '8521{enter}');
  expect(getByTestId('bingo').textContent).toBe('bingo ! game over ， replay?Yes');
  // expect(getByTestId('input')).toBeDisabled();
  // expect(input).toBeDisabled();
  // TypeError: expect(...).toBeDisabled is not a function
  expect(input.disabled).toBe(true);
  // 上面的input.disabled是參考網路的寫法 去抓dom裡面的屬性
  // expect(getByTestId('bingo')).toBeTruthy();
});

test('should yes', () => {
  const {getByTestId } = render(<HW />);
  userEvent.click(getByTestId('replay'));
  expect(getByTestId('ul').children.length).toBe(0);
});



