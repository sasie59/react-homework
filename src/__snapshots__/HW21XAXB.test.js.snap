// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should bingo Game Over & new Game  1`] = `
<div>
  <div>
    <h1>
      猜數字
    </h1>
    <ul>
      <li>
        一進到遊戲電腦隨機產生4位不重複數字
      </li>
      <li>
        使用者送出答案，若不符合「不重複4位數字」則跳錯誤訊息
      </li>
      <li>
        送出的答案跟正確答案比較，位置一樣則A，位置不同則B
      </li>
      <li>
        會累積過去猜過的答案與結果
      </li>
      <li>
        如果猜到 4A 則遊戲結束，並可以另開新局。
      </li>
    </ul>
    <form>
      <input
        type="text"
      />
      <button>
        猜!
      </button>
    </form>
    <div
      class="hide"
      data-testid="bingo"
    >
      bingo ! game over ， replay?
      <button
        data-testid="replay"
      >
        Yes
      </button>
    </div>
    <ul
      data-testid="ul"
    >
      <li
        data-testid="result-0"
      >
        8521: 1A1B
      </li>
    </ul>
  </div>
</div>
`;
